File: examportal/src/main/java/com/exam/service/CategoryService.java
package com.exam.service;

import java.util.Set;

import com.exam.model.Category;

public interface CategoryService{
	
	public Category addCategory(Category category);
	
	public Category updateCategory(Category category);
	
	public Category getCategory(Long categoryId);
	
	public Set<Category> getCategories();
	
	public void deleteCategory(Long categoryId);
	
}

File: examportal_frontend/exam-front/src/app/services/category.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import baseUrl from './helper';
import { Category } from '../model/category';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {
  constructor(private http:HttpClient ) { }


  public addCategory(category:Category) {
    return this.http.post(`${baseUrl}/category/`,category);
  }

  public updateCategory(category:Category) {
    return this.http.put(`${baseUrl}/category/`,category);
  }

  public getCategories() {
    return this.http.get(`${baseUrl}/category/`);
  }

  public deleteCategory(categoryId:number) {
   return this.http.delete(`${baseUrl}/category/${categoryId}`);
  }
}

